components/nexus-orient/src/main/java/org/sonatype/nexus/orient/entity/EntityAdapter.java

```diff
@@ -283,7 +283,7 @@ public AttachedEntityId(final ODocument document, final EntityAdapter owner) {
     public String getValue() {
       if (cachedValue == null) {
         final ORID identity = document.getIdentity();
-        checkState(!identity.isTemporary(), "attempted use of temporary document id");
+        checkState(!identity.isTemporary(), "attempted use of temporary document id - commit the transaction first");
```

Comment:
This was my change, thinking people wouldn't grasp the significance. I'll revise it.